lib/phoenix_live_view/colocated_js.ex

```diff
@@ -0,0 +1,337 @@
+  directory not specific to a single application.
+
+  The colocated JS directory follows this structure:
+
+  ```text
+  _build/$MIX_ENV/phoenix-colocated/
+  _build/$MIX_ENV/phoenix-colocated/my_app/
+  _build/$MIX_ENV/phoenix-colocated/my_app/index.js
+  _build/$MIX_ENV/phoenix-colocated/my_app/HASH_MyAppWeb.DemoLive/file.js
+  _build/$MIX_ENV/phoenix-colocated/my_dependency/HASH_MyDependency.Module/file.js
+  ...
+  ```
+
+  Each application has its own folder. Inside, each module also gets its own folder, which allows
+  us to track and clean up outdated code.
+
+  To use colocated JS from your `app.js`, your bundler needs to be configured to resolve the
+  `phoenix-colocated` folder. For new Phoenix applications, this configuration is already included
+  in the esbuild configuration inside `config.exs`:
+
+      config :esbuild,
+        ...
+        my_app: [
+          args:
+            ~w(js/app.js --bundle --target=es2022 --outdir=../priv/static/assets/js --external:/fonts/* --external:/images/* --alias:@=.),
+          cd: Path.expand("../assets", __DIR__),
+          env: %{
+            "NODE_PATH" => Enum.join([Path.expand("../deps", __DIR__), Mix.Project.build_path()], ":")
+          }
+        ]
+
+  The important part here is the `NODE_PATH` environment variable, which tells esbuild to also look
+  for packages inside the `deps` folder, as well as the `Mix.Project.build_path()`, which resolves to
+  `_build/$MIX_ENV`. If you use a different bundler, you'll need to configure it accordingly. If it is not
+  possible to configure the `NODE_PATH`, you can also change the folder to which LiveView writes colocated
+  JavaScript by setting the `:target_directory` option in your project's `config.exs`:
+
+  ```elixir
+  config :phoenix_live_view, Phoenix.LiveView.ColocatedJS,
+    target_directory: Path.expand("../assets/node_modules/phoenix-colocated", __DIR__)
+  ```
+
+  In this example, all colocated JavaScript would be written into the `assets/node_modules/phoenix-colocated`
+  folder.
+
+  ### Imports in colocated JS
+
+  The colocated JS files are fully handled by your bundler. For Phoenix apps, this is typically
+  `esbuild`. Because colocated JS is extracted to a folder outside the regular `assets` folder,
+  special care is necessary when you need to import other files inside the colocated JS:
+
+  ```javascript
+  import { someFunction } from "some-dependency";
+  import somethingElse from "@/vendor/vendored-file";
+  ```
+
+  While dependencies from `node_modules` should work out of the box, you cannot simply refer to your
+  `assets/vendor` folder using a relative path. Instead, your bundler needs to be configured to handle
+  an alias like `@` to resolve to your local `assets` folder. This is configured by default in the
+  esbuild configuration for new Phoenix applications using `esbuild`'s [alias option](https://esbuild.github.io/api/#alias).
```

Comment:
Let's add an example right below (or mention we already do it right above) and say that we include this by default from Phoenix v1.8 new apps?